subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/CompileClasspathPropertyAnnotationHandler.java

```diff
@@ -16,20 +16,41 @@
 
 package org.gradle.api.internal.project.taskfactory;
 
+import org.gradle.api.internal.TaskInternal;
 import org.gradle.api.internal.changedetection.state.CompileClasspathSnapshotter;
 import org.gradle.api.internal.changedetection.state.FileCollectionSnapshotter;
+import org.gradle.api.internal.tasks.TaskPropertyValue;
 import org.gradle.api.tasks.CompileClasspath;
+import org.gradle.api.tasks.CompileClasspathPropertyNormalizer;
+import org.gradle.api.tasks.InputFiles;
 
 import java.lang.annotation.Annotation;
 
-public class CompileClasspathPropertyAnnotationHandler extends ClasspathPropertyAnnotationHandler {
+public class CompileClasspathPropertyAnnotationHandler implements OverridingPropertyAnnotationHandler, FileSnapshottingPropertyAnnotationHandler {
     @Override
     public Class<? extends Annotation> getAnnotationType() {
         return CompileClasspath.class;
     }
 
     @Override
-    public Class<? extends FileCollectionSnapshotter> getSnapshotterType() {
+    public Class<? extends Annotation> getOverriddenAnnotationType() {
+        return InputFiles.class;
```

Comment:
Did we forget about that? 😱

Reply:
No. Previously this was inherited from `ClasspathPropertyAnnotationHandler`. I made the two classes independent in this PR, though.